nfs/controller.go

```diff
@@ -436,22 +442,30 @@ func (cs *CsiNfsService) ControllerUnpublishVolume(ctx context.Context, req *csi
 
 	service, slice, err := cs.getServiceAndSlice(ctx, serviceName)
 	if err != nil {
-		return resp, err
+		if strings.Contains(err.Error(), "not found") {
+			log.Infof("Endpoint slice or service might not exists: %v - slice or service might be deleted.", err)
+			return resp, nil
+		}
+
+		log.Errorf("failed to get service %s: %v", serviceName, err)
+		return nil, err
 	}
+
 	var nodeIPAddress string
 	if len(slice.Endpoints) > 0 {
 		address := slice.Endpoints[0]
 		nodeIPAddress = address.Addresses[0]
 	}
+
 	if nodeIPAddress == "" {
-		return resp, fmt.Errorf("endpointslice apparaently had no IP addresses %v", slice)
+		return nil, fmt.Errorf("endpointslice apparaently had no IP addresses %v", slice)
```

Comment:
spell check apparently, maybe even remove it. apparently does not give meaningful info here IMO